### DISCUSSION TITLE
libmesh installation error associated with compiling petscdmlibmeshimpl.C

### DISCUSSION THREAD

**Discussion by aeronova** [libMesh/libmesh#3502] [General]
Hello,

I was trying to install libmesh using WSL2 and got the following error while compiling src/solvers/petscdmlibmeshimpl.C. I'd appreciate it if someone can help me with the error. Please find the attached files (error messages and config) for your reference.

===============================

src/solvers/petscdmlibmeshimpl.C:881:59: warning: conversion lacks type at end of format [-Wformat=]
src/solvers/petscdmlibmeshimpl.C:881:59: warning: too many arguments for format [-Wformat-extra-args]
In file included from /home/aeronova/library/petsc/petsc-3.18.5/include/petscsys.h:1189:0,
                 from /home/aeronova/library/petsc/petsc-3.18.5/include/petscbag.h:5,
                 from /home/aeronova/library/petsc/petsc-3.18.5/include/petsc.h:5,
                 from ./include/libmesh/petsc_macro.h:66,
                 from src/solvers/petscdmlibmeshimpl.C:20:
src/solvers/petscdmlibmeshimpl.C: In function ‘PetscErrorCode DMSetUp_libMesh(DM)’:
src/solvers/petscdmlibmeshimpl.C:904:120: error: ‘struct _p_PetscObject’ has no member named ‘type’
 _SETERRQ2(((PetscObject)dm)->comm, PETSC_ERR_ARG_WRONG, "DM of type %s, not of type %s", ((PetscObject)dm)->type, DMLIB
                                                                                                             ^
/home/aeronova/library/petsc/petsc-3.18.5/include/petscerror.h:119:127: note: in definition of macro ‘SETERRQ’
  ierr_seterrq_petsc_ = PetscError(comm, __LINE__, PETSC_FUNCTION_NAME, __FILE__, ierr, PETSC_ERROR_INITIAL, __VA_ARGS__
                                                                                                             ^
src/solvers/petscdmlibmeshimpl.C:904:5: note: in expansion of macro ‘LIBMESH_SETERRQ2’
     LIBMESH_SETERRQ2(((PetscObject)dm)->comm, PETSC_ERR_ARG_WRONG, "DM of type %s, not of type %s", ((PetscObject)dm)->
     ^
Makefile:18873: recipe for target 'src/solvers/libmesh_dbg_la-petscdmlibmeshimpl.lo' failed
make[1]: *** [src/solvers/libmesh_dbg_la-petscdmlibmeshimpl.lo] Error 1
make[1]: Leaving directory '/home/aeronova/build/libmesh-1.7.1'
Makefile:32290: recipe for target 'all-recursive' failed
make: *** [all-recursive] Error 1

[libmesh_on_WSL2_installation_error.txt](https://github.com/libMesh/libmesh/files/11075219/libmesh_on_WSL2_installation_error.txt)
[config.log](https://github.com/libMesh/libmesh/files/11075220/config.log)

**Comment by aeronova** [comment]
I tired the same installation process, building petsc, slepc, and libmesh, on my MacBook Pro, and got an error while compiling the same file, libmesh_dbg_la-petscdmlibmeshimpl.lo. I have attached the configuration file for your reference.

[config.log](https://github.com/libMesh/libmesh/files/11085782/config.log)

**Reply by jwpeterson** [reply]
What version of libmesh are you using? For me line 904 of the file in question reads:
```
  if (!eq)
    LIBMESH_SETERRQ2(((PetscObject)dm)->comm, PETSC_ERR_ARG_WRONG, "DM of type %s, not of type %s", ((PetscObject)dm)->type_name, DMLIBMESH);
```

Note the `type_name` instead of `type`... it's likely that updating to at least e3ca53c384 (Oct 2022) will fix the compilation error for you.

**Reply by aeronova** [reply]
I'm using the latest release, version 1.7.1, which was released on Aug 9, 2022. So I guess I have to install the git version of libmesh (or older versions of petsc and slepc).
